src/System.IO.Pipes/tests/BaseCommonTests.cs

```diff
@@ -0,0 +1,312 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipes.Tests
+{
+    public abstract class BaseCommonTests
+    {
+        // used to make sure async calls don't return and throw proper exception
+        protected static void NotReachable(object obj) { Assert.True(false, "This should not be reached."); }
+
+        public static void ConnectedPipeWriteBufferNullThrows(PipeStream pipe)
+        {
+            Assert.True(pipe.IsConnected);
+            Assert.True(pipe.CanWrite);
+
+            Assert.Throws<ArgumentNullException>("buffer", () => pipe.Write(null, 0, 1));
+            Assert.Throws<ArgumentNullException>("buffer", () => NotReachable(pipe.WriteAsync(null, 0, 1)));
```

Comment:
I thought you were trying to make sure it threw on the 'call' side vs after waiting so that if the call returned (and potentially threw later), it would fail.